### PR TITLE
add input inputs autocomplete, id and name

### PR DESCRIPTION
**Pull request by zhen0**
## PR Checklist

(_all items must be checked off for a PR to be merged_)

This PR:

- [x] has a changelog entry with a short description of what's changed in `CHANGELOG.md`
- [x] adds new tests or updates existing tests (or hasn't, for reasons explained below)
- [x] strictly follows the design spec (if one exists)
- [x] adheres to [a11y guidelines](https://www.a11yproject.com/checklist/)/[WCAG](https://www.w3.org/WAI/standards-guidelines/wcag/) guidelines where applicable; this can be tested using Lighthouse (native in Chrome dev tools or as a plugin for other browsers) and with accessibility reports in Firefox

This PR has been tested in the following browsers (the latest version of each is fine):

- [x] Chrome
- [x] Firefox
- [x] Edge
- [x] Safari

### CODE REVIEW

File: src/components/Input/Input.vue [hunk 1]
```diff
@@ -134,6 +137,18 @@ export default defineComponent({
     pattern: {
       type: String,
       required: false
+    },
+    autocomplete: {
+      type: Boolean,
```
Comment (znicholasbrown) [inline src/components/Input/Input.vue:142]: This should be `type: string`; https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete

**Review by znicholasbrown — CHANGES_REQUESTED** [review]
I'm fine with adding these like this, though as @pleek91 mentioned in #173 it might make more sense to fully pass `$attrs` to the input node. Given that there are a bunch of props that are used for periphery options this might not make sense without a refactor but I'll defer to @pleek91 here

**Comment by pleek91** [comment]
@zhen0 @znicholasbrown I'm okay with adding these now for speed if needed but using `$attrs` is definitely the way to go. Otherwise we'll be back adding more attributes next time we need a different one

**Comment by zhen0** [comment]
> @zhen0 @znicholasbrown I'm okay with adding these now for speed if needed but using `$attrs` is definitely the way to go. Otherwise we'll be back adding more attributes next time we need a different one

I hadn't seen the suggestion in #173 but it definitely makes more sense.

File: src/components/Input/Input.vue [hunk 2]
```diff
@@ -145,7 +132,20 @@ export default defineComponent({
...
+      return this.disabled
+        ? ['disabled', ...baseList]
+        : this.invalid
+        ? ['invalid', ...baseList]
+        : this.active
+        ? ['active', ...baseList]
+        : this.hovered
+        ? ['hovered', ...baseList]
+        : baseList
+    },
```
Comment (pleek91) [inline src/components/Input/Input.vue:146]:
```suggestion
    outerClassList(): any {
      return [
        ...this.$attrs.class,
        {
          disabled: this.disabled,
          invalid: this.invalid,
          active: this.active,
          hovered: this.hovered
        }
      ]
    },
```

File: src/components/Input/Input.vue [hunk 3]
```diff
@@ -156,6 +156,10 @@ export default defineComponent({
         ? ['hovered']
         : []
     },
+    attributes() {
+      const { ...attrs } = this.$attrs
```
Comment (pleek91) [inline src/components/Input/Input.vue:160]: We need to remove `class` and `style` from `$attrs` so they don't get bound to the input.  
```suggestion
      const { class, style, ...attrs } = this.$attrs
```

**Review by pleek91 — CHANGES_REQUESTED** [review]

File: src/components/Input/Input.vue (hunk 2 above)
Comment (pleek91) [reply src/components/Input/Input.vue:146]: Realizing the logic here might not be quite right. The original will bind `disabled` but not `invalid` if you pass in both props. Which seems odd to me 🤔

**Review by pleek91 — APPROVED** [review]
LGTM!

**Review by znicholasbrown — APPROVED** [review]

**Pull request merged**